sway/desktop/output.c

```diff
@@ -412,6 +412,25 @@ static void render_container_simple_border_normal(struct sway_output *output,
 		render_texture(output->wlr_output, output_damage, title_texture,
 			&texture_box, matrix, 1.0);
 	}
+
+	// Marks
+	if (config->show_marks && marks_texture) {
+		float output_scale = output->wlr_output->scale;
+		struct wlr_box texture_box;
+		wlr_texture_get_size(marks_texture,
+			&texture_box.width, &texture_box.height);
+		texture_box.x = (box.x + box.width) * output_scale - texture_box.width;
+		texture_box.y = (box.y + box.height)
+			* output_scale - texture_box.height;
+
+		float matrix[9];
+		wlr_matrix_project_box(matrix, &texture_box,
+			WL_OUTPUT_TRANSFORM_NORMAL,
+			0.0, output->wlr_output->transform_matrix);
+
+		render_texture(output->wlr_output, output_damage, marks_texture,
```

Comment:
We should factor this into a function taking a texture and output-local coords.

Reply:
I don't think you can... If you look at the title one, we have to adjust the texture box between the calls to `wlr_matrix_project_box` and `render_texture`.

Maybe I'm misunderstanding you though, so correct me if I'm wrong.

Reply:
Yeah, with my patch merged in it's more complicated. We could pass the box instead, and force the box to be passed to `wlr_matrix_project_box` to have the size of the texture (to make sure the texture isn't improperly scaled).

Reply:
I still don't understand. If you want to refactor it, can you do it as a separate PR? It's not a blocker to this one.